### ISSUE TITLE
Wrong path referenced for failing test

### ISSUE THREAD

**Issue by felixriese** [pytest-dev/pytest#10168]
I locally moved my project folder `project/` to another directory from `/files/old/path/project` to `/files/new/project`. Then, I ran the tests with `pytest`, which works as expected - except the shown path to a test file when a test fails [output is censored]:

```bash
E   AssertionError: assert 'something' == <something_else'>
E     - something
E     ?          ^^
E     + something_else
E     ?          ^^^

/files/old/path/project/tests/test_something.py:30: AssertionError
```

When I run `pytest --vvv`, I get:

```bash
project/tests/test_something.py::TestName::test_something[SUCCESS-something] <- ../../project/tests/test_something.py FAILED [ 25%]
```
which shows the relative path to the *old* directory.

**Expected behavior:** The old directory is not shown anywhere.

---

`pyproject.toml`:
```toml
[tool.pytest.ini_options]
minversion = "7.1.2"
addopts = "--cov=app/ --cov-report html --cov-report term"
testpaths = [
    "tests"
]
required_plugins = ["pytest-cov", "pytest-mock"]
pythonpath = ["app"]
```

Setup on macOS 12.5 (M1):
```bash
platform darwin -- Python 3.9.7, pytest-7.1.2, pluggy-1.0.0 -- /Users/name/miniforge3/envs/env-name/bin/python3.9
cachedir: .pytest_cache
django: settings: backend.server.settings (from ini)
rootdir: /files/new/, configfile: pyproject.toml, testpaths: project/tests
plugins: anyio-3.5.0, env-0.6.2, dash-2.5.0, mock-3.8.2, django-4.5.2, cov-3.0.0
```

**Comment by felixriese** [comment]
Update:  Interestingly, this behavior can be observed only for **some** tests, the other are not linked to the old directory.

**Comment by felixriese** [comment]
Update 2: the issue is gone. Nothing has been changed. I cannot understand what happened.

(this issue can be closed if there is no value in discussing it. I would still keep it in GitHub since I might not be the only one with this issue)

**Comment by RonnyPfannschmidt** [comment]
Did you possibly rename the complete folder, pyc files cache the initial filename, so a stale byte code cache Could be a reason

Are any pytest options in use that may prevent the mismatch error that's supposed to happen then

**Comment by felixriese** [comment]
That could have been it. Thanks! 👍

**Comment by egor83** [comment]
I ran into a very similar problem; Ronny, thanks a lot for pointing me to the solution!

<details><summary>More details</summary>
<p>

Basically I am normally running tests locally against a DB in Docker; I ran tests via Docker once, and since then pytest stopped showing me traceback for the failed tests, only showing `>   ???` instead.
What seems to have happened is the following:
- running tests via Docker created Python cache (`__pycache__/*.pyc`) owned by root;
- re-running tests locally picked up that cached bytecode (either couldn't or didn't try to delete/overwrite it), tried accessing source code with the path hardcoded in cache (that is, with paths as they are in Docker), and failed.

Once I deleted cached files, it started working again.

</p>
</details>
